CoreAlgorithmPlugins/src/au/gov/asd/tac/constellation/plugins/algorithms/sna/similarity/CosineSimilarityPlugin.java

```diff
@@ -109,94 +110,132 @@ public void edit(final GraphWriteMethods graph, final PluginInteraction interact
 
         // map each vertex to its neighbour count
         final int vertexCount = graph.getVertexCount();
-        final int[][] neighbourWeights = new int[vertexCount][vertexCount];
-        final BitSet[] neighbours = new BitSet[vertexCount];
-        final BitSet update = new BitSet(vertexCount);
-        final BitSet selected = new BitSet(vertexCount);
-        for (int vertexPosition = 0; vertexPosition < vertexCount; vertexPosition++) {
-            final int vertexId = graph.getVertex(vertexPosition);
-            neighbours[vertexPosition] = new BitSet(vertexCount);
-            for (int vertexNeighbourPosition = 0; vertexNeighbourPosition < graph.getVertexNeighbourCount(vertexId); vertexNeighbourPosition++) {
-                final int neighbourId = graph.getVertexNeighbour(vertexId, vertexNeighbourPosition);
-                final int neighbourPosition = graph.getVertexPosition(neighbourId);
+        final List<VertexWithNeighbours> verticiesWithNeighbours = new ArrayList<>();
 
-                if (vertexPosition == neighbourPosition) {
-                    continue;
-                }
+        VertexWithNeighbours currentVertexWithNeighbour;
+        for (int vertexPosition = 0; vertexPosition < vertexCount; vertexPosition++) { //For each vertex
+            currentVertexWithNeighbour = null;
+            final int vertexId = graph.getVertex(vertexPosition);
+            
+            int vertexNeighbourCount = 0;
+            int bleh = graph.getVertexNeighbourCount(vertexId);
```

Comment:
😄

Reply:
haha yeah i've gone ahead and deleted it.

Reply:
Done